src/SudoSu.php

```diff
@@ -23,15 +23,27 @@ public function __construct(Application $app, AuthManager $auth, SessionManager
         $this->session = $session;
     }
 
-    public function loginAsUser($userId, $currentUserId)
+    /**
+     * Log the sudosu user in and preserve the original user id if exists
+     *
+     * @param string $sudosuUserId
+     * @param string|null $originalUserId
+     * @return void
+     */
+    public function loginAsUser($sudosuUserId, $originalUserId)
     {
         $this->session->put('sudosu.has_sudoed', true);
-        $this->session->put($this->sessionKey, $currentUserId);
+        $this->session->put($this->sessionKey, $originalUserId);
 
-        $this->auth->loginUsingId($userId);
+        $this->auth->loginUsingId($sudosuUserId);
```

Comment:
Great, thanks for the insight @krve. I'm going to rename the parameters back to what they were, but if you still think they cause confusion, feel free to open a separate PR and we can have a more in-depth discussion!